kotlinx-coroutines-core/common/src/Yield.kt

```diff
@@ -19,9 +19,8 @@ public suspend fun yield(): Unit = suspendCoroutineUninterceptedOrReturn sc@ { u
     val context = uCont.context
     context.checkCompletion()
     val cont = uCont.intercepted() as? DispatchedContinuation<Unit> ?: return@sc Unit
-    if (!cont.dispatcher.isDispatchNeeded(context)) {
-        return@sc if (cont.yieldUndispatched()) COROUTINE_SUSPENDED else Unit
-    }
+    // Special case for the unconfined dispatcher that can yield only in existing unconfined loop
```

Comment:
When squashing, please fix the type in KDoc: "but also check" -> "but also checks".

Reply:
I've squashed and totally updated KDoc for `yield` to better document its actual behavior.